packages/@react-aria/collections/src/BaseCollection.ts

```diff
@@ -254,7 +254,7 @@ export class BaseCollection<T> implements ICollection<Node<T>> {
       throw new Error('Cannot add a node to a frozen collection');
     }
 
-    if (node.type === 'item' && this.keyMap.get(node.key) == null) {
+    if (node.type === 'item' && this.keyMap.get(node.key) == null || node.type === 'header' && this.keyMap.get(node.key) == null) {
```

Comment:
updated so that we get the correct `aria-rowcount`, assuming that we want to count 'headers' as part of the rows